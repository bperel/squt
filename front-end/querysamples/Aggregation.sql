SELECT books.name as book_name, COUNT(DISTINCT chapters.name) AS nb_chapters
FROM books
INNER JOIN chapters ON books.id = chapters.book_id
GROUP BY books.name