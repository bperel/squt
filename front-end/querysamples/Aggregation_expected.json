{
	"Functions" : {
		"nb_chapters" : {
			"alias" : "nb_chapters",
			"name" : "COUNT_DISTINCT_FUNC",
			"group" : "1",
			"to" : "OUTPUT"
		}
	},
	"Tables" : {
		"books" : {
			"books" : {
				"OUTPUT" : {
					"name" : {
						"-1" : "book_name"
					}
				},
				"GROUP" : {
					"name" : "ASC"
				},
				"CONDITION" : {
					"id" : {
						"JOIN" : {
							"chapters.book_id" : "JOIN_TYPE_STRAIGHT"
						}
					}
				}
			}
		},
		"chapters" : {
			"chapters" : {
				"OUTPUT" : {
					"name" : {
						"nb_chapters" : "name"
					}
				}
			}
		}
	}
}